test/writer/test_writer.py

```diff
@@ -36,21 +39,26 @@ def test_write() -> None:
 
 
 def test_anonymous_gate() -> None:
-    register_manager = RegisterManager(QubitRegister(2))
-    ir = IR()
-    ir.add_gate(CR(Qubit(0), Qubit(1), Float(1.234)))
-    ir.add_gate(BlochSphereRotation(Qubit(0), axis=(1, 1, 1), angle=1.23))
-    ir.add_gate(CR(Qubit(0), Qubit(1), Float(1.234)))
-    circuit = Circuit(register_manager, ir)
-
+    qc = CircuitBuilder(2, 2)
+    qc.H(Qubit(0))
+    qc.ir.add_gate(BlochSphereRotation(Qubit(0), axis=(1, 1, 1), angle=1.23))
+    qc.ir.add_gate(ControlledGate(Qubit(0), BlochSphereRotation(Qubit(0), axis=(1, 1, 1), angle=1.23)))
+    qc.ir.add_gate(MatrixGate(np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]]), [Qubit(0), Qubit(1)]))
+    qc.CR(Qubit(0), Qubit(1), Float(1.234))
     assert (
-        writer.circuit_to_string(circuit)
+        writer.circuit_to_string(qc.to_circuit())
         == """version 3.0
 
 qubit[2] q
+bit[2] b
 
-CR(1.234) q[0], q[1]
-<anonymous-gate>
+H q[0]
+BlochSphereRotation(Qubit[0], axis=[0.57735 0.57735 0.57735], angle=1.23, phase=0.0)
+ControlledGate(control_qubit=Qubit[0], BlochSphereRotation(Qubit[0], axis=[0.57735 0.57735 0.57735], angle=1.23, phase=0.0))
+MatrixGate(qubits=[Qubit[0], Qubit[1]], matrix=[[1 0 0 0]
```

Comment:
In order to abide by the CQASM standards, we may want to have our commands inline, even if it won't be parsed through CQASM.

Reply:
Agreed! I've made sure that it outputs to a single line.